src/lib/storage/table.cpp

```diff
@@ -18,75 +19,71 @@
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
+Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size(target_chunk_size) { create_new_chunk(); }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  // We only allow adding columns for empty tables to avoid dealing with default values
+  Assert(row_count() == 0, "Adding a column is only allowed for empty tables");
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  resolve_data_type(type, [&_chunks = _chunks](auto type) {
+    using Type = typename decltype(type)::type;
+    _chunks.back()->add_segment(std::make_shared<ValueSegment<Type>>());
+  });
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_chunks.back()->size() >= _target_chunk_size) {
+    create_new_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto new_chunk = std::make_shared<Chunk>();
+  _chunks.push_back(new_chunk);
+  for (const auto& type : _column_types) {
+    resolve_data_type(type, [&new_chunk](auto type) {
+      using Type = typename decltype(type)::type;
+      new_chunk->add_segment(std::make_shared<ValueSegment<Type>>());
+    });
+  }
 }
 
-ColumnCount Table::column_count() const {
-  // Implementation goes here
-  return ColumnCount{0};
-}
+ColumnCount Table::column_count() const { return ColumnCount{_column_names.size()}; }
 
 ChunkOffset Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  // Calculate the count of all full chunks multiplied by the target/max chunk size
+  // and then add the size of the last, incomplete chunk
+  // Since our knowledge about chunks is still limited, we assume that this is the right
+  // method for now. This is up to change later.
+  return ChunkOffset{(chunk_count() - 1) * _target_chunk_size + _chunks.back()->size()};
```

Comment:
Complementary to the above comment, chunks can have different sizes. For sprint 1, however, this is fine.